File: C++/739. Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> ans(n, 0);
        stack<pair<int, int>> s;
        for(int i=0;i<n;i++)
        {
            while(!s.empty() && s.top().second < temperatures[i])
            {
                int index = s.top().first;
                s.pop();
                ans[index] = i-index;
            }
            s.push(make_pair(i, temperatures[i]));
        }
        return ans;
    }
};

File: Python/739. Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        s = list()
        ans = [0] * len(temperatures)
        for i in range(len(temperatures)):
            while len(s) != 0 and s[-1][1] < temperatures[i]:
                ans[s[-1][0]] = i- s[-1][0]
                s.pop()
            s.append([i, temperatures[i]])
        return ans